pkg/repository/git.go

```diff
@@ -196,3 +252,25 @@ func (ogr oktetoGitWorktree) Status(ctx context.Context, localGit LocalGitInterf
 
 	return oktetoGitStatus{status: status}, nil
 }
+
+func (ogr oktetoGitWorktree) BuildContextStatus(ctx context.Context, localGit LocalGitInterface, buildContext string) (oktetoGitStatus, error) {
+	// using git directly is faster, so we check if it's available
+	_, err := localGit.Exists()
+	if err != nil {
+		// git is not available, so we fall back on git-go
+		oktetoLog.Debug("Calculating git status: git is not installed, for better performances consider installing it")
+		status, err := ogr.worktree.Status()
```

Comment:
How does this know if this is referred for a specific context. I can't see clear where it's specifying it

Reply:
I mean using the external dependency

Reply:
yes, we need to filter only relevant status related to the service context

Reply:
d6723152baa1fda86f4cf0a53fdfb4da9db1613e